### PR TITLE
Feature/upgrade python 3 8

### PR DESCRIPTION
**Pull request by fivejjs**
upgrade python version

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
+FROM opendatacube/geobase-builder:${V_BASE} as env_builder
 ENV LC_ALL=C.UTF-8
 
 # Install our Python requirements
-COPY docker/requirements.txt docker/version.txt /conf/
+COPY docker/requirements.txt docker/version.txt docker/constraints.txt /conf/
 
 RUN cat /conf/version.txt && \
   env-build-tool new /conf/requirements.txt ${py_env_path}
```
Comment (alexgleith) [inline Dockerfile:11]: I think you need to add constraints here.
```suggestion
  env-build-tool new /conf/requirements.txt /conf/constraints.txt ${py_env_path}
```

File: Dockerfile [hunk 2]
```diff
@@ -18,17 +19,24 @@ RUN /env/bin/pip install \
   rm -rf /tmp/dea_ml
```
Comment (alexgleith) [inline Dockerfile:19]: ??

File: Dockerfile [hunk 3]
```diff
@@ -18,17 +19,24 @@ RUN /env/bin/pip install \
...
 
 # Below is the actual image that does the running
-FROM opendatacube/geobase:runner
+FROM opendatacube/geobase-runner:${V_BASE}
 ARG py_env_path=/env
 
+# Environment can be whatever is supported by setup.py
+# so, either deployment, test
+ARG ENVIRONMENT=deployment
+RUN echo "Environment is: $ENVIRONMENT"
```
Comment (alexgleith) [inline Dockerfile:28]: Why is this here?

File: Dockerfile [hunk 4]
```diff
@@ -18,17 +19,24 @@ RUN /env/bin/pip install \
...
+
 ENV DEBIAN_FRONTEND=noninteractive \
     PATH="${py_env_path}/bin:${PATH}" \
     LC_ALL=C.UTF-8 \
     LANG=C.UTF-8
 
+RUN apt-get update \
+    && apt-get install software-properties-common -y \
+    && apt-get upgrade -y
```
Comment (alexgleith) [inline Dockerfile:37]: Not a good idea, I don't think.

File: docker/requirements.txt [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
-pyyaml==5.3.1
-botocore==1.15.32
+aiobotocore==1.3.1
+pyyaml==5.4.1
+botocore==1.20.49
 s3transfer==0.3.7
-boto3==1.12.30
-GDAL
+boto3==1.17.49
+GDAL==3.3.0
```
Comment (alexgleith) [inline docker/requirements.txt:14]: I think you should remove all specific requirements. Requirements are set with the constraints file now.

**Review by alexgleith — CHANGES_REQUESTED** [review]
You've included some unrelated code changes to the Dockerfile changes, I think.

You're also not using the constraints file, so let's fix that!

File: Dockerfile (hunk 2 above)
Comment (fivejjs) [reply Dockerfile:19]: This is from the original I didn't touch.

File: Dockerfile (hunk 3 above)
Comment (fivejjs) [reply Dockerfile:28]: copy from the alchemist docker file to indicate the environment during building.

File: Dockerfile (hunk 4 above)
Comment (fivejjs) [reply Dockerfile:37]: otherwise the libbooks and libcgal  can not be found

File: production/dea_ml/dea_ml/core/predict_from_feature.py [hunk 6]
```diff
@@ -39,13 +38,9 @@ def predict_with_model(model, data, td_url) -> xr.Dataset:
     # load the column names from the
     # training data file to ensure
     # the bands are in the right order
-    response = requests.get(td_url)
-    with open("td.txt", "w") as f:
-        f.write(response.text)
-    with open("td.txt", "r") as file:
+    with fsspec.open(td_url, "r") as file:
```
Comment (cbur24) [inline production/dea_ml/dea_ml/core/predict_from_feature.py:41]: @fivejjs totally approve of this!

**Review by alexgleith — APPROVED** [review]
